rest-service/manager_rest/rest/resources_v3_1/filters.py

```diff
@@ -51,22 +54,84 @@ def put(self, filter_id):
         Currently, this function only supports the creation of a labels filter
         """
         rest_utils.validate_inputs({'filter_id': filter_id})
-        request_dict = rest_utils.get_json_and_verify_params(
-            {'filter_rules': {'type': list}})
-        labels_filters = request_dict['filter_rules']
-        parsed_labels_filters = _parse_labels_filters(labels_filters)
-        visibility = rest_utils.get_visibility_parameter(
-            optional=True, valid_values=VisibilityState.STATES)
-
+        parsed_labels_filters, visibility = _get_filter_rules_and_visibility()
+        now = get_formatted_timestamp()
         new_filter = models.Filter(
             id=filter_id,
             value=json.dumps({'labels': parsed_labels_filters}),
-            created_at=get_formatted_timestamp(),
+            created_at=now,
+            updated_at=now,
             visibility=visibility
         )
 
         return get_storage_manager().put(new_filter)
 
+    @authorize('filters_get')
+    @rest_decorators.marshal_with(models.Filter)
+    def get(self, filter_id, _include=None):
+        """
+        Get a filter by ID
+        """
+        return get_storage_manager().get(
+            models.Filter, filter_id, include=_include)
+
+    @authorize('filters_update')
+    def delete(self, filter_id):
+        """
+        Delete a filter by ID
+        """
+        storage_manager = get_storage_manager()
+        filter_elem = storage_manager.get(models.Filter, filter_id)
+        _validate_filter_modification_permitted(filter_elem)
+        storage_manager.delete(filter_elem, validate_global=True)
+        return None, 204
+
+    @authorize('filters_update')
+    @rest_decorators.marshal_with(models.Filter)
+    def patch(self, filter_id):
+        """Update a filter by its ID
+
+        This function updates the filter rules and visibility
+        """
+        if not request.json:
+            raise manager_exceptions.IllegalActionError(
+                'Update a filter request must include at least one parameter '
+                'to update')
+
+        labels_filters, visibility = _get_filter_rules_and_visibility(True)
+        storage_manager = get_storage_manager()
+        filter_elem = storage_manager.get(models.Filter, filter_id)
+        if visibility:
+            get_resource_manager().validate_visibility_value(
+                models.Filter, filter_elem, visibility)
+            filter_elem.visibility = visibility
+        if labels_filters:
+            filter_elem.value = json.dumps({'labels': labels_filters})
```

Comment:
This will replace the entire filter rules, without excluding the same rules from the list. What do you think?